nightshift/bot.py

```diff
@@ -1,27 +1,121 @@
 """
 Launches NightShift application
 """
-
+import argparse
 import logging
 import logging.config
 import loggly.handlers
+import os
+import sys
+
+from sqlalchemy.exc import IntegrityError
+import yaml
 
+from nightshift import datastore_transactions, manager
 from nightshift.config.logging_conf import log_conf
 
-# from nightshift.manager import process_resources
 
-conf = log_conf()
-logging.config.dictConfig(conf)
-logger = logging.getLogger("nightshift")
+def config_local_env_variables(
+    config_file: str = "nightshift/config/config.yaml",
+) -> None:
+    """
+    Sets up environment variables to run NighShift locally.
+    Requires a valid config.yaml file in the nighshift/config directory
+
+    Args:
+        config_file:            path to config file that includes environmental
+                                variables
+    """
+    with open(config_file, "r") as f:
+        data = yaml.safe_load(f)
+        for k, v in data.items():
+            os.environ[k] = v
+
+
+def configure_database(env: str = "prod") -> None:
+    """
+    Sets up proper database tables and populates them with constant data
+
+    Args:
+        env:                    environment to set up database
+    """
+    if env == "local":
+        config_local_env_variables()
+    try:
+        datastore_transactions.init_db()
+        print(f"NightShift {env} database successfully set up.")
+    except IntegrityError as exc:
```

Comment:
@mwbenowitz 
Yes, `IntegrityError` is thrown when trying to setup existing database. It happened to me, so I thought it may be a good idea to include it.
I added a block of code (commit https://github.com/BookOps-CAT/NightShift/pull/14/commits/8bd2148ce149ddca7d4228936897cc28f2155a4c) that asserts certain required elements (checks if tables are there, values for libraries, certain resource categories, etc.) and added additional `AssertionError` exception in the `bot.configure_database` module to trap them.
I think that maybe it would be better to move the entire validation into a separate function, but I would like to wait with that. I thought about your advise to use [database as a sole authoritative source of values](https://github.com/BookOps-CAT/NightShift/issues/12) instead of reusing info from `constants.py` where these were hard coded to prepopulate the database. I will heed your advice which will require creation of a few additional tables, etc. After completion of that element it will be maybe a better time make changes here.